firefox_puppeteer/api/software_update.py

```diff
@@ -0,0 +1,544 @@
+            return active_update.patchCount;
+
+        """)
+
+    @property
+    def selected_patch_final_url(self):
+        """Get the finalURL of the selected patch for the active update
+
+        :returns: The patch size
+        """
+
+        return self.marionette.execute_script("""
+            let ums = Cc['@mozilla.org/updates/update-manager;1']
+                .getService(Ci.nsIUpdateManager);
+            let active_update = ums.activeUpdate;
+            return active_update.selectedPatch.finalURL;
+
+        """)
+
+    @property
+    def selected_patch_size(self):
+        """Get the size of the selected patch for the active update
+
+        :returns: The patch size
+        """
+
+        return self.marionette.execute_script("""
+            let ums = Cc['@mozilla.org/updates/update-manager;1']
+                .getService(Ci.nsIUpdateManager);
+            let active_update = ums.activeUpdate;
+            return active_update.selectedPatch.size;
+
+        """)
+
+    @property
+    def selected_patch_type(self):
+        """Get the type of the selected patch for the active update
+
+        :returns: The patch type
+        """
+
+        return self.marionette.execute_script("""
+            let ums = Cc['@mozilla.org/updates/update-manager;1']
+                .getService(Ci.nsIUpdateManager);
+            let active_update = ums.activeUpdate;
+            return active_update.selectedPatch.type;
+
+        """)
+
+
+class MARChannels(object):
+    """Class to handle the allowed MAR channels as listed in update-settings.ini
+    """
+
+    INI_SECTION = 'Settings'
+    INI_OPTION = 'ACCEPTED_MAR_CHANNEL_IDS'
+
+    REGEX_UPDATE_CHANNEL_PREF = re.compile(r'("app\.update\.channel", ")([^"].*)(?=")')
+
+    def __init__(self, software_update):
+
+        self.marionette = software_update.marionette
+        self.prefs = software_update.prefs
+
+        self.ini_file_path = self.marionette.execute_script("""
+            Cu.import('resource://gre/modules/Services.jsm');
+
+            var file = Services.dirsvc.get('GreD', Ci.nsIFile);
+            file.append('update-settings.ini');
+
+            return file.path;
+
+            """)
+
+    @property
+    def config(self):
+        config = ConfigParser.ConfigParser()
+        config.readfp(open(self.ini_file_path))
+        return config
+
+    @property
+    def config_file_path(self):
+        return self.ini_file_path
+
+    @property
+    def config_file_contents(self):
+        with open(self.ini_file_path) as f:
+            return f.read()
+
+    @property
+    def channels(self):
+        channels = self.config.get(self.INI_SECTION, self.INI_OPTION)
+        return channels.split(',')
+
+    def set_channels(self, channels):
+        channels = ','.join(channels)
+        new_config = self.config
+        new_config.set(self.INI_SECTION, self.INI_OPTION, channels)
+        with open(self.ini_file_path, 'wb') as f:
+            new_config.write(f)
```

Comment:
Have you checked that this produces the correct output? When I was implementing this class originally I have seen that the Python config library added blanks around the `=` sign for this key. This will break reading the ini file in Firefox.